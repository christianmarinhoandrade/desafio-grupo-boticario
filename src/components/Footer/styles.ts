import { Layout, Typography } from 'antd'
import styled from 'styled-components'

import { FooterProps } from './index'

const FooterAntd = Layout.Footer
const TextAntd = Typography.Text

export const Footer = styled(FooterAntd)`
  background-color: transparent;
  text-align: center;
`

export const Text = styled(TextAntd)<FooterProps>`
  color: ${(props) => props.color};
`
