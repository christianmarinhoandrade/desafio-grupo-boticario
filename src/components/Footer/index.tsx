import * as S from './styles'

export type FooterProps = {
  color?: string
}

function SharedFooter(props: FooterProps) {
  const { color } = props

  return (
    <S.Footer>
      <S.Text color={color}>© </S.Text>
      <S.Text strong color={color}>
        Grupo Boticário
      </S.Text>
      <S.Text color={color}> | {new Date().getFullYear()}</S.Text>
    </S.Footer>
  )
}

export default SharedFooter
