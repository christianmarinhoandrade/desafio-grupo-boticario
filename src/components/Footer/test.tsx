import { render, screen } from '@testing-library/react'

import Footer from '.'

describe('<Footer />', () => {
  it('should render footer default', () => {
    const { container } = render(<Footer />)

    expect(screen.getByText(/grupo boticário/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render footer with color', () => {
    render(<Footer color="#fff" />)

    expect(screen.getByText(/grupo boticário/i).parentElement).toHaveStyle({
      color: '#fff'
    })
  })

  it('should render footer without color', () => {
    render(<Footer />)

    expect(screen.getByText(/grupo boticário/i).parentElement).not.toHaveStyle({
      color: '#fff'
    })
  })
})
