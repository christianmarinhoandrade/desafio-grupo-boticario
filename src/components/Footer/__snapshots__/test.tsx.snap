// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render footer default 1`] = `
.c0 {
  background-color: transparent;
  text-align: center;
}

<footer
  class="ant-layout-footer c0"
>
  <span
    class="ant-typography "
  >
    © 
  </span>
  <span
    class="ant-typography "
  >
    <strong>
      Grupo Boticário
    </strong>
  </span>
  <span
    class="ant-typography "
  >
     | 
    2021
  </span>
</footer>
`
